lib/beaker-hostgenerator/generator.rb

```diff
@@ -104,6 +112,19 @@ def __parse_node_info_token(token)
         node_info['arbitrary_roles'] = []
       end
 
+      if node_info['host_settings']
+        # Turns a string like "{foo=bar,this=that}" into a
+        # hash like {'foo' => 'bar', 'this' => 'that'}.
+        node_info['host_settings'] = Hash[
+          node_info['host_settings'].
+          delete('{}').
+          split(',').
+          map { |keyvalue| keyvalue.split('=') }
+        ]
+      else
+        node_info['host_settings'] = {}
+      end
+
       return node_info
     end
```

Comment:
@nwolfe right -- I was thinking about `generate` and got a bit cross-eyed -- missed the method boundary.

This is definitely a good tidbit for ESO CodeCraft, if we wanted to pull the original version up on the screen and talk about ways to refactor / extract, and the tradeoffs between the tests we have, the tests we'd want (what those even might be), and what could / should be done to get there.